packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts

```diff
@@ -121,6 +120,45 @@ export class InsuredBridgeL2Client {
     });
   }
 
+  async getEventsForBlockSearch(blockSearchConfig: { fromBlock: number; toBlock: number }): Promise<EventData[][]> {
+    const [fundsDepositedEvents, whitelistedTokenEvents] = await Promise.all([
+      this.bridgeDepositBox.getPastEvents("FundsDeposited", blockSearchConfig),
+      this.bridgeDepositBox.getPastEvents("WhitelistToken", blockSearchConfig),
+    ]);
+
+    // If there is a fallback L2 web3 provider, check that the deposit box events are also found by those providers,
+    // otherwise throw an error. This allows the caller to have additional confidence about the accuracy of fetched L2
+    // state.
+    const fundsDepositedTransactionHashes = fundsDepositedEvents.map((event) => event.transactionHash);
+    const whitelistedTokenTransactionHashes = whitelistedTokenEvents.map((event) => event.transactionHash);
+    for (let i = 0; i < this.fallbackL2Web3s.length; i++) {
+      const _bridgeDepositBox = (new this.fallbackL2Web3s[i].eth.Contract(
+        getAbi("BridgeDepositBox"),
+        this.bridgeDepositAddress
+      ) as unknown) as BridgeDepositBoxWeb3;
+      const [_fundsDepositedEvents, _whitelistedTokenEvents] = await Promise.all([
+        _bridgeDepositBox.getPastEvents("FundsDeposited", blockSearchConfig),
+        _bridgeDepositBox.getPastEvents("WhitelistToken", blockSearchConfig),
+      ]);
+      _fundsDepositedEvents.forEach((event) => {
+        if (!fundsDepositedTransactionHashes.includes(event.transactionHash)) {
+          throw new Error(
```

Comment:
Are we sure we want to _throw_? Assuming most of the event data is valid and agreed upon, would this cause us to block the relayer/disputer from running even on non-contested items?

Reply:
Hm what should we do if not throw? Should we return the union of all events found?

Reply:
that looks great! makes sence to throw error & log.